doc/perf.md

```diff
@@ -236,22 +228,16 @@ Segmentation fault
 $
 ```
 
-But with optimizations enabled, the program does not crash even if I type `y`:
+But with optimizations enabled, the program does not crash even when typing `y`:
 
 ```
 $ ./jou -O3 asd.jou
 Crash this program? (y/n) y
 $
 ```
 
-The optimizer assumes that you don't attempt to access the value of a `NULL` pointer.
-In other words, it thinks that the `x = *foo` code will never run,
-and it can therefore be ignored.
-
-Correctly written Jou code does not break when optimizations are enabled.
-For example, accessing a `NULL` pointer simply isn't a good way to exit a program;
-your program shouldn't ever do it.
-If you want the program to crash, you can use the `abort()` function, for example:
+The optimizations make the program ignore the code to access the value of a `NULL` pointer.
+Essentially it thinks that the `x = *foo` code will never run, because you aren't supposed to access the value of a NULL pointer. This code will thus get ignored. Sidenote: if you want the program to crash with optimizations on, then you should do so using `abort()` function for example:
```

Comment:
```suggestion
Essentially it thinks that the `x = *foo` code will never run, because you aren't supposed to access the value of a NULL pointer. This code will thus get ignored.

Sidenote: if you want the program to crash with optimizations on, then you should do so using `abort()` function for example:
```